### PR TITLE
feat(_tests): re-enabling the app tests

### PR DESCRIPTION
**Pull request by arschles**

### CODE REVIEW

**Comment by arschles** [comment]
@mboersma I am still unable to successfully run integration tests (even with the ones in this PR disabled) locally so I could use some help. Meanwhile I'll rebase these and see if they're ready to go afterward.

**Comment by mboersma** [comment]
@arschles once I got router's domain annotations updated so I could see the workflow controller from outside the cluster, tests have been running well for me (not everything passes, though):
```console
$ DEIS_WORKFLOW_SERVICE_HOST=deis.10.245.1.3.xip.io make test-integration
```

File: _tests/apps_test.go [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
 
 import (
+	"fmt"
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/config"
 	. "github.com/onsi/gomega"
 )
 
+func getRandAppName() string {
+	return fmt.Sprintf("apps-test-%d", GinkgoConfig.RandomSeed)
```
Comment (bacongobbler) [inline _tests/apps_test.go]: won't this return the same value both times, since the seed is generated at execution time?

Comment (bacongobbler) [inline _tests/apps_test.go]: we could use this as a seed to a `rand` generator, for example:

```
func init() {
    rand.Seed(GinkgoConfig.RandomSeed)
}

var letterRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")

func RandStringRunes(n int) string {
    b := make([]rune, n)
    for i := range b {
        b[i] = letterRunes[rand.Intn(len(letterRunes))]
    }
    return string(b)
}
```

Then we can just use `fmt.Sprintf("apps-test-%d", RandStringRunes(5))`.

Comment (arschles) [inline _tests/apps_test.go]: Yes, good call that I totally glossed over. We can keep it simple and stick with a random int here. Changes incoming.

**Comment by arschles** [comment]
@mboersma I can successfully hit the controller, through the router, manually with the `deis` CLI using `$DEIS_WORKFLOW_SERVICE_HOST:$DEIS_WORKFLOW_SERVICE_PORT`. For example:

```console
ENG000656:workflow aaronschlesinger$ echo $DEIS_WORKFLOW_SERVICE_HOST
deis.example.com
ENG000656:workflow aaronschlesinger$ echo $DEIS_WORKFLOW_SERVICE_PORT
80
ENG000656:workflow aaronschlesinger$ deis register http://$DEIS_WORKFLOW_SERVICE_HOST:$DEIS_WORKFLOW_SERVICE_PORT
username: arschles1
password: 
password (confirm): 
email: ar@b.com
Registered arschles1
Logged in as arschles1
```

Are the test failures in [this gist](https://gist.github.com/arschles/d1e6ced9a0d4292b70aa) similar to what you see?

**Comment by mboersma** [comment]
I've seen similar failures. One thing that trips up the tests is that (like the v1 tests), they expect a clean slate, so that the first user registered gets admin privileges. If a user isn't successfully deleted, then the next test run will fail on anything requiring admin privs.

We need to fix any case where the cleanup didn't happen, but for now I've been getting inside deis/workflow and deleting the model objects to ensure my next test run gets somewhere:
```console
$ kubectl exec -it deis-workflow-qoxhz -- sh
/app # ./manage.py shell
Python 2.7.10 (default, Aug 13 2015, 12:27:27) 
[GCC 4.9.2] on linux2
>>> from django.contrib.auth import get_user_model
>>> m = get_user_model()
>>> m.objects.exclude(username='AnonymousUser').delete()
>>> m.objects.all()                                     
[<User: AnonymousUser>]
```

**Pull request merged**